edb/server/net_worker.py

```diff
@@ -382,7 +387,8 @@ async def gc(server: edbserver.BaseServer) -> None:
             if tenant.accept_new_tasks
         ]
         try:
-            await asyncio.wait(tasks)
+            if tasks:
+                await asyncio.wait(tasks)
```

Comment:
Why aren't we using a `TaskGroup` here?

Reply:
Yeah, I guess we can; this was a carried-over from [an old commit](https://github.com/edgedb/edgedb/pull/6233/commits/47c3c09973adf08c20c7917b7a836ad26ddaa387), let me try again so that we know for sure.

Reply:
Ah it's just unnecessary, because the Task we're in here is safe without cancellation - the server doesn't need to wait for `gc()` itself to clean up anything, while all child Tasks are already managed by the `TaskGroup` in each Tenant (by `tenant.create_task()`).

We would need an extra coroutine wrapper to use `TaskGroup` here:

```python
async def _await_task(task: asyncio.Task):
    return await task


async def gc(server: edbserver.BaseServer) -> None:
    while True:
        try:
            async with asyncio.TaskGroup() as g:
                for tenant in server.iter_tenants():
                    if tenant.accept_new_tasks:
                        task = tenant.create_task(
                            _gc(tenant, NET_HTTP_REQUEST_TTL),
                            interruptable=False,
                        )
                        g.create_task(_await_task(task))
...
```